packages/database-integrations/src/database-integration-env-vars.ts

```diff
@@ -466,7 +467,13 @@ const getTrinoEnvVars = (
   integrationId: string,
   metadata: DatabaseIntegrationMetadataByType['trino'],
   projectRootDirectory: string
-): SqlAlchemyInput => {
+): SqlAlchemyInput | null => {
+  // OAuth is a federated auth method, handled separately
+  if (metadata.authMethod === TrinoAuthMethods.Oauth) {
+    return null
+  }
+
+  // At this point, metadata must be password auth (either with authMethod: 'password' or authMethod: null)
```

Comment:
nit - maybe we should make it explicit that `authMethod: null` is for backwards compatibility here as well